stdlib/3.4/statistics.pyi

```diff
@@ -0,0 +1,44 @@
+# Stubs for statistics
+# See: http://docs.python.org/3/library/statistics.html
+
+import sys
+from typing import Iterable, Iterator, Sequence, TypeVar, Union, overload
+from decimal import Decimal
+
+# Note: according to the docs, statistics only explicitly supports
+# int, float, Decimal, and Fraction. Other types within the numeric
+# tower are not supported. It also states mixing together different
+# types results in undefined behavior, so the type signatures below
+# deliberately enforce that numeric types may not be mixed.
+#
+# TODO: Once either https://github.com/python/typeshed/pull/545 or
+# https://github.com/python/typeshed/pull/94 is accepted, add support
+# for fractions.
+_TNum = TypeVar('_TNum', int, float, Decimal)
+_T = TypeVar('_T')
+
+def mean(data: Union[Iterator[_TNum], Sequence[_TNum]]) -> _TNum: ...
```

Comment:
Oh, hmm -- when I did `mean([1, 2, 3])`, I got back an int, but after some poking, that turned out to be only because 1 + 2 + 3 is actually evenly divisible by 3.

Perhaps I could get rid of int within the `TypeVar` and rely on mypy to understand that ints should be auto-promoted to `float`?

Regarding mixing ints with Decimals/mixing floats with Decimals: the [documentation for statistics](https://docs.python.org/3/library/statistics.html) stated that mixing together different types is "undefined and implementation-dependent", and even recommends running `map(float, input_data)` first if you have mixed numeric types.

So, if mixing together things like Decimal and int is undefined, then I think it's ok and actually for the best if typeshed forbids that altogether?